### PR TITLE
Add Hooks Example

### PR DESCRIPTION
**Pull request by MistaTwist**

### CODE REVIEW

File: src/webparts/helloWorld/components/HelloWorld.tsx [hunk 1]
```diff
@@ -9,8 +9,14 @@ export interface IHelloWorldProps {
...
 }
 
-export const HelloWorld: React.FunctionComponent<IHelloWorldProps> = (props) =>
-  <>
+export const HelloWorld: React.FunctionComponent<IHelloWorldProps> = (props) => {
+  const [count, setCount] = React.useState(0);
+
+  React.useEffect(() => {
+    document.getElementById("timesClicked").innerText = `You clicked ${count} times`;
+  });
```
Comment (garrytrinder) [inline src/webparts/helloWorld/components/HelloWorld.tsx]: `useEffect` is generally preferred for fetching data using Hooks as a replacement for `componentDidMount` life-cycle method and I don't think that it is a good idea to be changing the DOM outside of React.

If we were pulling the initial state value of `count` from an API then `useEffect` would be a good approach, I would suggest that this effect is removed for now as you are using a Hook to load initial state and updating via the `setCount` method.

File: src/webparts/helloWorld/components/HelloWorld.tsx [hunk 2]
```diff
@@ -23,6 +29,11 @@ export const HelloWorld: React.FunctionComponent<IHelloWorldProps> = (props) =>
             </a>
           </div>
         </div>
+        <div className={styles.row}>
+          <PrimaryButton text="Click Me" onClick={() => setCount(count + 1)} />
+          <div id="timesClicked"></div>
```
Comment (garrytrinder) [inline src/webparts/helloWorld/components/HelloWorld.tsx]: As the effect to update the DOM is to be removed, I would update this element to include the text and `count` variable.

Comment (MistaTwist) [reply src/webparts/helloWorld/components/HelloWorld.tsx]: Thanks - I've made those corrections. Do they now show up ok?

**Comment by garrytrinder** [comment]
Manually merged. Thank you 🎉

**Pull request closed without merging**